### PR TITLE
Add Exception manager DELETE THIS ONE

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context
We need an exception manager to help us view and skip unprocessable messages with zero downtime.

# What has changed
Added the Exception Manager.

# How to test?
Build. Run. Publish bad message. Check the logs. Check the Exception Manager.

# Links
Trello: https://trello.com/c/2rkYNAMa

### CODE REVIEW

**Comment by NickGrantONS** [comment]
[Exception Manager v1.0](https://trello.com/c/2rkYNAMa/1266-exception-manager-v10)

File: src/main/java/uk/gov/ons/census/casesvc/service/ReceiptService.java [hunk 1]
```diff
@@ -29,10 +32,18 @@ public void processReceipt(ResponseManagementEvent receiptEvent) {
...
+
+    if (caze != null) {
+      caze.setReceiptReceived(true);
+      caseService.saveAndEmitCaseUpdatedEvent(caze);
+    } else {
+      log.with("qid", receiptPayload.getQuestionnaireId())
+          .with("tx_id", receiptEvent.getEvent().getTransactionId())
+          .with("channel", receiptEvent.getEvent().getChannel())
+          .warn("Receipt received for unaddressed UAC/QID pair not yet linked to a case");
+    }
```
Comment (ryangrundy7) [inline src/main/java/uk/gov/ons/census/casesvc/service/ReceiptService.java]: Is this meant to be in here? looks like stuff for decoupling response_received and questionnaire_linked

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/casesvc/service/ReceiptService.java]: yeah that's in production. We don't want to regress a fix we put in.

File: src/main/java/uk/gov/ons/census/casesvc/utility/QuestionnaireTypeHelper.java [hunk 2]
```diff
@@ -58,8 +52,8 @@ public static int calculateQuestionnaireType(String treatmentCode) {
           return 34;
       }
     } else {
-      log.with("treatment_code", treatmentCode).error(UNEXPECTED_CASE_TYPE_ERROR);
-      throw new IllegalArgumentException();
+      throw new IllegalArgumentException(
+          String.format("Unexpected Case Type fortreatment code '%s'", treatmentCode));
```
Comment (ryangrundy7) [inline src/main/java/uk/gov/ons/census/casesvc/utility/QuestionnaireTypeHelper.java]: Need a space in the log line

**Comment by NickGrantONS** [comment]
Mistake with merging. Have recreated a new branch

**Pull request closed without merging**